src/dialog/error.cpp

```diff
@@ -0,0 +1,149 @@
+/**
+    gTox a GTK-based tox-client - https://github.com/KoKuToru/gTox.git
+
+    Copyright (C) 2015  Luca Béla Palkovics
+    Copyright (C) 2014  Maurice Mohlek
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>
+**/
+#include "error.h"
+#include <glibmm/i18n.h>
+#include <glibmm/markup.h>
+#include <execinfo.h>
+#include <cxxabi.h>
+
+using namespace dialog;
+
+error::error(Gtk::Window& parent,bool fatal,std::string title, std::string message):
+    MessageDialog(parent, title, false, fatal?Gtk::MESSAGE_ERROR:Gtk::MESSAGE_WARNING, Gtk::BUTTONS_CLOSE, true) {
+    set_secondary_text(((fatal)?_("ERROR_REPORT_TEXT_INTRO"):"")
+                       + Glib::Markup::escape_text(message)
+                       + ((fatal)?(Glib::Markup::escape_text("\n\n" + get_stacktrace())
+                                   + _("ERROR_REPORT_TEXT_OUTRO")):""), true);
+
+    if (fatal) {
+        add_button(_("ERROR_REPORT_BTN"), 213);
```

Comment:
Just checked documentation any positive number is okay,
It's used to check which button got pressed. at line 57.

I could replace it with a constexpr..